optimize/listsearch.go

```diff
@@ -0,0 +1,119 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// ListSearch finds the optimum location from a specified list of possible
+// optimum locations. The dimension specified in init must match the number
```

Comment:
s/init/InitGlobal/
Although I'm not sure whether these comments wouldn't better go to InitGlobal and Locs.